subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DependencyServices.java

```diff
@@ -38,21 +38,25 @@ public void registerGradleUserHomeServices(ServiceRegistration registration) {
         registration.addProvider(new DependencyManagementGradleUserHomeScopeServices());
     }
 
+    @Override
     public void registerBuildSessionServices(ServiceRegistration registration) {
-        registration.addProvider(new Object() {
-            CacheLockingManager createCacheLockingManager(CacheRepository cacheRepository, ArtifactCacheMetaData artifactCacheMetaData) {
-                return new DefaultCacheLockingManager(cacheRepository, artifactCacheMetaData);
-            }
-
-            TransformedFileCache createTransformedFileCache(ArtifactCacheMetaData artifactCacheMetaData, CacheRepository cacheRepository, InMemoryCacheDecoratorFactory cacheDecoratorFactory, FileSystemSnapshotter fileSystemSnapshotter, ListenerManager listenerManager) {
-                DefaultTransformedFileCache transformedFileCache = new DefaultTransformedFileCache(artifactCacheMetaData, cacheRepository, cacheDecoratorFactory, fileSystemSnapshotter);
-                listenerManager.addListener(transformedFileCache);
-                return transformedFileCache;
-            }
-        });
+        registration.addProvider(new DependencyManagementBuildSessionServices());
```

Comment:
This diff looks unrelated to your changes?

Reply:
At one point, I had added something to these services, so it was getting ugly to keep the anonymous class around.